src/DssSpell.sol

```diff
@@ -18,6 +18,53 @@ pragma solidity 0.8.16;
 
 import "dss-exec-lib/DssExec.sol";
 import "dss-exec-lib/DssAction.sol";
+import "dss-interfaces/dss/IlkRegistryAbstract.sol";
+import "dss-interfaces/dss/GemJoinAbstract.sol";
+import "dss-interfaces/ERC/GemAbstract.sol";
+
+interface Initializable {
+    function init(bytes32 ilk) external;
+}
+
+interface RwaLiquidationLike {
+    function ilks(bytes32) external view returns (string memory doc, address pip, uint48 tau, uint48 toc);
+    function init(bytes32 ilk, uint256 val, string calldata doc, uint48 tau) external;
+}
```

Comment:
Good question, any thoughts on that? I used `string calldata doc` as I found this in the archive. But if you look on code itself you will see that this is calldata and note memory. Maybe foundry issue???

Reply:
> Good question, any thoughts on that?

actually not really, i'm far from being awesome at smart contract development-support.

https://github.com/makerdao/spells-mainnet/blob/b34e641fbc634468e2e95620daf30bee80e808e4/archive/2022-12-09-DssSpell/DssSpell.sol#L54

`calldata` seems to be the way to go. let's leave it as is here because it worked in the past. I'm not sure why this difference is there, but i also don't think there's time right now to spend debugging/investigating it.

P.S. 
tagging my comrade for the sake of making this case here more noticeable for him as well @SidestreamColdMelon

Reply:
If you look at source code here: https://etherscan.io/address/0x88f88Bb9E66241B73B84f3A6E197FbBa487b1E30#code

It has `function init(bytes32 ilk, uint256 val, string calldata doc, uint48 tau)`

Reply:
Pretty sure it's a foundry issue

Reply:
Hmmm, I was checking forge docs, and when you provide an address as an argument to `cast interface`, it actually fetches the ABI from Etherscan and builds the interface from there.

If you get the ABI for [0x362dfE51E4f91a8257B8276435792095EE5d85C3](https://goerli.etherscan.io/address/0x362dfE51E4f91a8257B8276435792095EE5d85C3#code
) you can find the following there:


```json
  {
    "constant": false,
    "inputs": [
      {
        "internalType": "bytes32",
        "name": "ilk",
        "type": "bytes32"
      },
      {
        "internalType": "uint256",
        "name": "val",
        "type": "uint256"
      },
      {
        "internalType": "string",
        "name": "doc",
        "type": "string"
      },
      {
        "internalType": "uint48",
        "name": "tau",
        "type": "uint48"
      }
    ],
    "name": "init",
    "outputs": [],
    "payable": false,
    "stateMutability": "nonpayable",
    "type": "function"
  },
```

Notice how there is no `memory` or `calldata` specifier there, because [it's not part of the ABI](https://docs.soliditylang.org/en/v0.8.19/abi-spec.html#mapping-solidity-to-abi-types).